.github/workflows/checks.yml

```diff
@@ -22,7 +22,20 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
-      - uses: actions/setup-node@v3
+      - uses: actions/setup-node@v4
       - uses: ./.github/actions/security/lockfile
         with:
           allowedHosts: 'yarn'
+
+  install-integrity:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v4
+        with:
+          node-version: 20.x
+      - name: Monorepo install
+        uses: ./.github/actions/yarn-nm-install
+      - name: 👬🏽 Check for duplicated deps
+        run: yarn dedupe --check
```

Comment:
Will display the list and fail if there is duplicates

Reply:
what do you foresee as the process for individuals to solve this check? run the command?

Reply:
In my experience yarn 3+ already dedupes very well. 

Contributors that would add a dependency in their pull requests won’t never trigger a ci check error here.

But if it does fail:  ie the contributor edited the lock manually or because of a yarn issue,  he will have to run the command: yarn dedupe to generate a new lock file then push it.

There’s sense to add an howto in the contributor guideline. But it will be rare.

in the other hand tools that directly modifies the lock might trigger a failure Dependabot..,

hope it clarifies

Reply:
I have a question you can probably answer :)

If dependabot tends to cause dupes, what do you feel would be the best strategy for using it? Is there a way to trick it into behaving better? or will someone see the error and have to know they need to manually pull the PR and dedupe it before merging?

I like this idea but we'll need to set some sort of guide/policy for "when you see this failed test on dependabot, you need to take these specific steps" so that people don't get lost and confused.

Reply:
Indeed. My personal journey into that was to move away from dependabot (and in general whatever directly change the lock file). Not sure about dependabot recent evolutions, but still seems

https://github.com/dependabot/dependabot-core/issues/5830

There’s ways to circumvent that directly in the ci, cause we can detect the user but it’s weird at the end.

I’d recommend to renovate instead. You’ll find a note in the issue description.

otherwise yes documentation or add a comment in the ci step to direct the people to an how-to


Another option is to not check for duplicate at all in the ci. An run a schedule action every week that just dedupe and create a pr if there’s lock change.